### ISSUE TITLE
Support validation of graphql input syntax

### ISSUE THREAD

**Issue by mikesol** [meeshkan/purescript-graphql-validator#1]
This package currently only supports validating that a result received from a graphql server corresponds to a schema. To be complete, it should also validate that the request sent to a server corresponds to a schema.

**Comment by aby2s** [comment]
Why do we need to validate something that we generate in runtime?

**Comment by fornwall** [comment]
@aby2s Asked mike the same on slack, and it's for catching bugs in the generator.

**Comment by mikesol** [comment]
Exactly. This is one of the unfortunate areas where we need to duplicate work a bit with other packages that have been written - there is an excellent graphql validator in purescript already, and there are also validators in python and javascript. However, because our ecosystem of graphql packages makes such heavy use of one AST, we need things like parsers, validators, and generators that work with that AST. We could of course write a bridge from our AST to another AST and do the validation that way but it is just as much work as creating our own.
